targets/TARGET_Realtek/TARGET_AMEBA/rtw_emac.cpp

```diff
@@ -217,7 +217,7 @@ emac_interface_t *wlan_emac_init_interface()
 {
 
     if (_emac == NULL) {
-        _emac = new emac_interface_t();
+        _emac = (emac_interface_t*) malloc(sizeof(emac_interface_t));
```

Comment:
This should use new or error in the case it fails, not print

Reply:
@sg- Everywhere in file print statements are used instead error, I can add that as bug and get it fix in new PR

Reply:
new would have failed as assert but malloc won't. That is the change in behavior.

Reply:
Oh, @deepikabhavnani, this was the issue with const members right?

@sg-, the `emac_interface_t` as written can't be default constructed in c++ ([here](https://github.com/ARMmbed/mbed-os/blob/master/hal/emac_api.h#L150)). The const member ([ops](https://github.com/ARMmbed/mbed-os/blob/master/hal/emac_api.h#L151)) must be constructed but the `emac_interface_t` class doesn't have a constructor to initialize the member. Since the `emac_api` header is also included by C files, we can't add a constructor. C supports struct initializers for this case, but C++ relies on constructors.

We can either remove the const or allocate with malloc. We could change the error print to assert.